### ISSUE TITLE
Figure out why scrolling doesn't work in some terminals

### ISSUE THREAD

**Issue by klange** [klange/bim#8]
I'm pretty sure these terminals support the scroll escapes, but maybe they require setting a region first.

**Comment by honestly-nuts** [comment]
What terminals have you tested bim on? And one which terminals do they not work?

**Comment by honestly-nuts** [comment]
I meant on

**Comment by klange** [comment]
Linux console doesn't support the scroll escapes, and I had issues with tmux in the past. Linux _does_ support row insertion and removal, which has been the preferred method for a while now and should probably be enabled there...

**Comment by honestly-nuts** [comment]
By Linux console do you mean gnome-terminal?

**Comment by klange** [comment]
No, I mean the Linux console.

**Comment by honestly-nuts** [comment]
The virtual console? I don't think that even supports scrolling.

**Comment by Henry-Wilson** [comment]
I have never encountered a tty/vt that supports any type of mouse action. It's not technically impossible, and there are many, many tty/vt options to choose from (getty, agetty, etc...). In practice, though, mouse input almost universally depends on X11/Wayland servers for handling.

To @klange's point, I have also had issues scrolling with bim in tmux. I don't believe I've ever tried to scroll in tty/vt. Let me check that I'm not mistaken... as expected, I can get no mouse input to register in any way.

At any rate, I've never tried to make bim work in tmux. It works fine in st and in alacritty, and I tend to think that tmux either has an option to pass along mouse commands (that I don't know about), or is failing in some capacity to meet with APIs common to X11 - which, frankly, as a vt-bound program, it has a right to do.

I just never worried about it. Are there any graphical (X11/Wayland) terminals that support scrolling/mouse input with other software, but not with bim?

Anyways, love bim. I can't live without my modal hjkl's, and I see no reason to use software more complicated than this. It's simple and robust, meeting every need I have in a code editor.

**Comment by klange** [comment]
The problem this ticket about is _sending_ a scroll command, which tells the terminal to shift the displayed view region. They are CSI sequences for `S` and `T`. Linux console and some other terminals _do_ support something else: `L` and `M` insert and delete rows in the terminal, respectively, and shift the rest of the content as necessary, which is effectively the same thing, so I've switched to that because it seems to be more widely supported. `S` and `T` scrolling are still available where supported.

On the topic of _mouse_ support: tmux, screen, any VTE terminal, etc. all support sequences bim expects. The Linux console _can_ support a mouse through a daemon called `gpm`, but it works differently. Correctly built versions of Vim usually have support for that setup, but I haven't really looked into it as I don't spend any time on the native console.

Anyway, having found that the terminals that originally spurred this ticket don't support the escape sequence I was using, and having fixed at least the one I cared about to use a different escape sequence, this investigation and ticket are done.
